src/Agent.Worker/ScriptJobExtension.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Agent.Worker.Handlers;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker
+{
+    // This class is for internal testing and is not publicly supported. This will be removed from the agent at a later time.
+    public class ScriptJobExtension : AgentService
```

Comment:
in your Powershell script, you will get the accesstoken via environment variable INPUT_ACCESSTOKEN. 
I don't want you use PowerShellHandler since it's a deprecated handler.